test/convert/mutt_get_content_info.c

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ * Test code for mutt_ch_convert_string()
+ *
+ * @authors
+ * Copyright (C) 2019 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define TEST_NO_MAIN
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "config.h"
+#include "acutest.h"
+#include "mutt/lib.h"
+#include "config/charset.h"
+#include "config/set.h"
+#include "config/types.h"
+#include "core/neomutt.h"
+#include "convert/convert.h"
+#include "convert_common.h"
+#include "test_common.h"
+
+static struct ConfigDef CharsetVars[] = {
+  // clang-format off
+  { "assumed_charset", DT_SLIST|SLIST_SEP_COLON|SLIST_ALLOW_EMPTY, 0, 0, charset_slist_validator,
+    "If a message is missing a character set, assume this character set"
+  },
+  { "charset", DT_STRING|DT_NOT_EMPTY|DT_CHARSET_SINGLE, 0, 0, charset_validator,
+    "Default character set for displaying text on screen"
+  },
+  { "config_charset", DT_STRING, 0, 0, charset_validator,
+    "Character set that the config files are in"
+  },
+  { "send_charset", DT_SLIST|SLIST_SEP_COLON|SLIST_ALLOW_EMPTY|DT_CHARSET_STRICT, IP "us-ascii:iso-8859-1:utf-8", 0, charset_slist_validator,
+    "Character sets for outgoing mail"
+  },
+  { NULL },
+  // clang-format on
+};
+
+void test_mutt_get_content_info(void)
+{
+  // struct Content *mutt_get_content_info(const char *fname, struct Body *b, struct ConfigSubset *sub);
+
+  char *fname = malloc(L_tmpnam);
+  TEST_CHECK(tmpnam_r(fname) != NULL);
```

Comment:
Sorry, FreeBSD isn't happy with `tmpnam_r()`
Our `mutt_file_mkstemp()` function unlinks the temp file, but you could cut'n'paste the rest

Reply:
Oops, missed that failure.

Reply:
So I actually just changed this to `tmpnam()` which is POSIX (while `tmpnam_r` isn't), and the difference as far as I can tell is only when passing `NULL` to the function, which I'm not doing.

Anyway, the builds are passing, so I think that should be fine.